### PR TITLE
fix: corrected bedrock model list

### PR DESCRIPTION
**Pull request by zhaoninge**
litellm.model_list and litellm.model_cost are not the correct list of model ids. 
Such as, bedrock/us-west-2/mistral.mixtral-8x7b-instruct-v0:1 should be bedrock/mistral.mixtral-8x7b-instruct-v0:, without the rengion name.
Now fix for AWS bedrock model, using the official sdk to get the model id in real time.
Before using it, you need to configure AWS_REGION_NAME AWS_ACCESS_KEY_ID and AWS_SECRET_ACCESS_KEY.

### CODE REVIEW

**Comment by li-boxuan** [comment]
Question: any reason this fix isn't targeting litellm repo?

**Review by Umpire2018 — COMMENTED** [review]
1. I think @li-boxuan  are right, this issue should be raised with the `litellm` repository. 
2. Even if `litellm` does not use the latest list of llms, this will not affect the operation of opendevin. I personally don't think it's necessary to introduce specific modifications to the codebase for this issue. If we do make such modifications, what about other similar issues?

**Comment by rbren** [comment]
TBH I'm thinking we should redo this logic--there are way too many model names, and it's really easy to pick the wrong one (e.g. `azure/gpt-4-turbo` instead of `gpt-4-turbo`)

I generally agree we should depend on LiteLLM upstream and ask them to fix. I do like having the AWS settings though

**Comment by zhaoninge** [comment]
I agree with everyone.
I have sent a request to litellm to provide an API that can query available models in real time. (https://github.com/BerriAI/litellm/issues/361, so many wishes)
However, until A responds, I think this fix can be used as a temporary solution, otherwise we can't use the AWS bedrock model.
Sorry, I can only partially solve the model id problem of AWS bedrock with my current energy. If we agree, we can work together to solve the problems of other providers.

**Comment by rbren** [comment]
I'm good with this one as-is. But I'm going to open a ticket to simplify the dropdown here, so we only have a handful of options to pick from, maybe separated by provider.

File: opendevin/llm/bedrock.py [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+from opendevin.schema import ConfigType
+
+AWS_ACCESS_KEY_ID = config.get(ConfigType.AWS_ACCESS_KEY_ID)
+AWS_SECRET_ACCESS_KEY = config.get(ConfigType.AWS_SECRET_ACCESS_KEY)
+AWS_REGION_NAME = config.get(ConfigType.AWS_REGION_NAME)
+
+
+def list_foundation_models():
+    try:
+        client = boto3.client(service_name='bedrock',
```
Comment (li-boxuan) [inline opendevin/llm/bedrock.py:14]: We shouldn't bother even calling this API, if `AWS_ACCESS_KEY_ID` and `AWS_SECRET_ACCESS_KEY` are None.

**Review by rbren — APPROVED** [review]

File: opendevin/core/config.py [hunk 2]
```diff
@@ -23,6 +23,9 @@
     ConfigType.LLM_API_KEY: None,
     ConfigType.LLM_BASE_URL: None,
     ConfigType.LLM_CUSTOM_LLM_PROVIDER: None,
+    ConfigType.AWS_ACCESS_KEY_ID: None,
+    ConfigType.AWS_SECRET_ACCESS_KEY: None,
+    ConfigType.AWS_REGION_NAME: 'us-west-2',
```
Comment (rbren) [inline opendevin/core/config.py:28]:
```suggestion
    ConfigType.AWS_REGION_NAME: None
```

File: opendevin/core/config.py [hunk 3]
```diff
@@ -23,6 +23,9 @@
     ConfigType.LLM_API_KEY: None,
     ConfigType.LLM_BASE_URL: None,
     ConfigType.LLM_CUSTOM_LLM_PROVIDER: None,
+    ConfigType.AWS_ACCESS_KEY_ID: None,
+    ConfigType.AWS_SECRET_ACCESS_KEY: None,
+    ConfigType.AWS_REGION_NAME: None
```
Comment (li-boxuan) [inline opendevin/core/config.py:28]:
```suggestion
    ConfigType.AWS_REGION_NAME: None,
```

**Review by li-boxuan — APPROVED** [review]

**Pull request merged**